server/ServerFSM.cpp

```diff
@@ -1138,7 +1138,14 @@ sc::result MPLobby::react(const JoinGame& msg) {
     Networking::ClientType client_type;
     std::string client_version_string;
     boost::uuids::uuid cookie;
-    ExtractJoinGameMessageData(message, player_name, client_type, client_version_string, cookie);
+    try {
+        ExtractJoinGameMessageData(message, player_name, client_type, client_version_string, cookie);
+    } catch (const std::exception& e) {
+        ErrorLogger(FSM) << "MPLobby::react(const JoinGame& msg): couldn't extract data from join game message";
+        player_connection->SendMessage(ErrorMessage(UserString("ERROR_INCOMPATIBLE_VERSION"), true));
```

Comment:
I don't believe `ERROR_INCOMPATIBLE_VERSION` is the correct message here.  The client could also be broken or act malicious.  So unless you can identify the version of the client reliable I would rather suggest to name it like `ERROR_BAD_REQUEST` in style of the HTTP 4XX client error response codes (to be exact `HTTP 400 - Bad Request`).

Reply:
sure, sounds reasonable...